### ISSUE TITLE
Use a taxon database to provide suggested entries in the taxa editor

### ISSUE THREAD

**Issue by robyngit** [NCEAS/metacatui#2090]
- This would involve an API call to do taxon lookup in the UI (similar to what we do with the NSF lookup for awards)
- There are several taxonomic data services we might use, like [WORMS](https://www.marinespecies.org/about.php). Some are more complete than others in some domains. (We will likely want these services configurable in the AppModel)
- We used to support a feature like this in Morpho
- Implementing this feature in a more generalizable way would allow use to use it in other places in the editor where we could do API lookups against vocabularies and external services.

Requested by the [SCTLD hosted repo](https://sctld.dataone.org/). Related to https://github.com/NCEAS/metacatui/issues/2086 and other open issues related to the taxa editor [here](https://github.com/NCEAS/metacatui/issues?q=is%3Aopen+is%3Aissue+label%3A%22taxa+editor%22)

**Comment by yvanlebras** [comment]
Yes, an important feature! We integrated it in our MetaShARK R Shiny app reusing EML Assembly Line R package with ncbi, worms, gbif taxonomic references

**Comment by robyngit** [comment]
Thanks @yvanlebras, those are useful details! Also good to know that there are multiple repositories interested in this feature.
